modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -546,39 +732,130 @@ ADLAROMANCEStressUpdateBase::computePolynomial(const ADReal & value,
+      _num_tiles,
+      std::vector<std::vector<std::vector<Real>>>(
+          _num_outputs, std::vector<std::vector<Real>>(_num_inputs, std::vector<Real>(3))));
 
-  for (unsigned int i = 0; i < _num_outputs; ++i)
+  for (unsigned int t = 0; t < _num_tiles; ++t)
   {
-    for (unsigned int j = 0; j < _num_inputs; ++j)
+    for (unsigned int i = 0; i < _num_outputs; ++i)
     {
-      for (unsigned int k = 0; k < 2; ++k)
+      transformed_limits[t][i] = limits[t];
+      for (unsigned int j = 0; j < _num_inputs; ++j)
       {
-        if (_transform[i][j] == ROMInputTransform::EXP)
-          transformed_limits[i][j][k] = std::exp(_input_limits[j][k] / _transform_coefs[i][j]);
-        else if (_transform[i][j] == ROMInputTransform::LOG)
-          transformed_limits[i][j][k] = std::log(_input_limits[j][k] + _transform_coefs[i][j]);
-        else
-          transformed_limits[i][j][k] = _input_limits[j][k];
+        for (unsigned int k = 0; k < 2; ++k)
+          convertValue(
+              transformed_limits[t][i][j][k], _transform[t][i][j], _transform_coefs[t][i][j]);
+        transformed_limits[t][i][j][2] =
+            (transformed_limits[t][i][j][1] - transformed_limits[t][i][j][0]) / 2.0;
       }
     }
   }
 
   return transformed_limits;
 }
 
-std::vector<std::vector<unsigned int>>
+std::vector<unsigned int>
 ADLAROMANCEStressUpdateBase::getMakeFrameHelper() const
 {
-  std::vector<std::vector<unsigned int>> v(_num_coefs, std::vector<unsigned int>(_num_inputs));
+  std::vector<unsigned int> v(_num_coefs * _num_inputs);
 
   for (unsigned int numcoeffs = 0; numcoeffs < _num_coefs; ++numcoeffs)
     for (unsigned int invar = 0; invar < _num_inputs; ++invar)
-      v[numcoeffs][invar] = numcoeffs / MathUtils::pow(_degree, invar) % _degree;
+      v[numcoeffs + _num_coefs * invar] = numcoeffs / MathUtils::pow(_degree, invar) % _degree;
 
   return v;
 }
+
+void
+ADLAROMANCEStressUpdateBase::computeStressFinalize(const ADRankTwoTensor & plastic_strain_increment)
+{
+  _cell_dislocation_increment = 0.0;
+  _wall_dislocation_increment = 0.0;
+
+  precomputeROM(_cell_output_index);
+  for (unsigned int t = 0; t < _num_tiles; ++t)
+    if (_weights[t])
+      _cell_dislocation_increment += _weights[t] * computeROM(t, _cell_output_index);
+
+  precomputeROM(_wall_output_index);
+  for (unsigned int t = 0; t < _num_tiles; ++t)
+    if (_weights[t])
+      _wall_dislocation_increment += _weights[t] * computeROM(t, _wall_output_index);
+
+  _cell_rate[_qp] = _cell_dislocation_increment / _dt;
+  _wall_rate[_qp] = _wall_dislocation_increment / _dt;
+  _cell_dislocations[_qp] = _old_input_values[_cell_output_index] + _cell_dislocation_increment;
+  _wall_dislocations[_qp] = _old_input_values[_wall_output_index] + _wall_dislocation_increment;
+
+  for (unsigned int i = 0; i < _num_inputs; i++)
+    checkInputWindow(_input_values[i], _window_failure[i], _global_limits[i]);
```

Comment:
Again for a later merge request, in the simulation I ran with the new P91 model the check didn't seem to make a difference here for the dislocation densities that were below the bounds. Definitely user error on my part with not setting all the flags could have been an issue.

That said, considering an exception to cut the time step or something similar if the dislocations are outside of the evolution ranges might be prudent here

Reply:
did you set `cell_input_window_failure` to the default value of `error`? That may have been the problem if you were using an old version of the code?

Reply:
I did not; I just ran with the defaults. I will add that option into my input files. thanks

Reply:
Well thats a bit werid since the default is already error though

Reply:
well petunias

Reply:
Try running with verbose=true to see if the global windows make sense

Reply:
Oh, I am already running with that option, and yes, the global values make sense. I double checked those values against the values from the getInputLimits method and they matched. 

I'll push up the demo input file and move that discussion over there